shared/utils/logging.hpp

```diff
@@ -1,402 +1,80 @@
 #pragma once
 
-#include <android/log.h>
-#include <pthread.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <dlfcn.h>
-#include <unistd.h>
-#include <cmath> // Included to support cmath's definition of log
 #include <string_view>
-#include <algorithm>
-#include <string>
-#include <list>
-#include <mutex>
-#include "modloader/shared/modloader.hpp"
-#include "utils-functions.h"
-#include <thread>
-#include <unordered_set>
-#include <unordered_map>
+#include <fmt/format.h>
+#include <fmt/ranges.h>
 
-namespace Logging {
-    enum Level {
-        CRITICAL = ANDROID_LOG_FATAL,
-        ERROR = ANDROID_LOG_ERROR,
-        WARNING = ANDROID_LOG_WARN,
-        INFO = ANDROID_LOG_INFO,
-        DEBUG = ANDROID_LOG_DEBUG
-    };
-}
+#include "paper/shared/logger.hpp"
+#include "paper/shared/modinfo_fmt.hpp"
+
+#include "utils-functions.h"
 
-#ifdef log
-#undef log
-#endif
+#include "il2cpp-api-types.h"
+#include "il2cpp-class-internals.h"
 
-#ifndef LOG_PATH
-#define LOG_PATH "/sdcard/Android/data/%s/files/logs/"
-#endif
+namespace il2cpp_utils {
+    Paper::LoggerContext const& getLogger();
+    std::string ClassStandardName(const Il2CppClass* klass, bool generics);
+    const char* TypeGetSimpleName(const Il2CppType* type);
+}
 
```

Comment:
General comment for most of the following formatters: They do NOT null check! This is incredibly important as a null should still be formatted correctly (or as correctly as possible) without crashing. Ex, a null class should say that it is null, for example.